src/mon/OSDMonitor.cc

```diff
@@ -7493,6 +7493,8 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     ss << "will thrash map for " << thrash_map << " epochs";
     ret = thrash();
     err = 0;
+    if (ret)
```

Comment:
in addition to "what", could you put the motive of this change into the commit, say, "why"?

also, `wait_for_finished_proposal()` does not trigger an update.